src/NHibernate/Driver/DriverExtensions.cs

```diff
@@ -36,7 +36,7 @@ public static DbTransaction BeginTransaction(this IDriver driver, IsolationLevel
 			var beginMethod = driver.GetType().GetMethod(
 				nameof(DriverBase.BeginTransaction),
 				new[] { typeof(IsolationLevel), typeof(DbConnection) });
-			if (beginMethod != null)
+			if (beginMethod != null && beginMethod.ReturnType == typeof(DbTransaction) && !beginMethod.IsStatic)
```

Comment:
`beginMethod.IsStatic` should not be here. `GetMethod` has an overload to specify the flags of the method you want to find. (Hint: `BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic`).

Reply:
I know, there are just a lot more parameters to these overloads, so I have been lazy.